#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Barry DeFrese adds weak_aliases to setresgid and setresuid,
#		   fixing undefined references when e.g. building putty.
# DP: Related bugs: #295118: libc0.3-dev: undefined references to setres{g,u}id
# DP: Dpatch author: Michael Banck <mbanck@debian.org>, GOTO Masanori <gotom@debian.org>
# DP: Patch author: 
# DP: Upstream status: In CVS
# DP: Status Details: 
# DP: Date: 2005-04-10

PATCHLEVEL=0

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p$PATCHLEVEL < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p$PATCHLEVEL < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
2005-02-10  Barry deFrese  <bddebian@comcast.net>

	* sysdeps/mach/hurd/setresgid.c: Add weak alias.
	* sysdeps/mach/hurd/setresuid.c: Add weak alias.

Index: setresgid.c
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/mach/hurd/setresgid.c,v
retrieving revision 1.1
diff -u -r1.1 setresgid.c
--- sysdeps/mach/hurd/setresgid.c	14 Oct 2002 01:03:11 -0000	1.1
+++ sysdeps/mach/hurd/setresgid.c	13 Feb 2005 20:08:02 -0000
@@ -1,5 +1,5 @@
 /* setresgid -- set effective group ID, real group ID, and saved-set group ID
-   Copyright (C) 2002 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2005 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -75,3 +75,4 @@
   return err;
 }
 libc_hidden_def (__setresgid)
+weak_alias (__setresgid, setresgid)
Index: setresuid.c
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/mach/hurd/setresuid.c,v
retrieving revision 1.1
diff -u -r1.1 setresuid.c
--- sysdeps/mach/hurd/setresuid.c	14 Oct 2002 01:03:11 -0000	1.1
+++ sysdeps/mach/hurd/setresuid.c	13 Feb 2005 20:08:02 -0000
@@ -1,5 +1,5 @@
 /* setresuid -- set effective user ID, real user ID, and saved-set user ID
-   Copyright (C) 2002 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2005 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -75,3 +75,4 @@
   return err;
 }
 libc_hidden_def (__setresuid)
+weak_alias (__setresuid, setresuid)
